daemon/emer-boot-id-provider.c

```diff
@@ -280,3 +280,42 @@ emer_boot_id_provider_get_id (EmerBootIdProvider *self,
   id_is_valid = TRUE;
   return TRUE;
 }
+
+#define KERNEL_CMDLINE_PATH	"/proc/cmdline"
+#define LIVE_BOOT_FLAG_REGEX	"\\bendless\\.live_boot\\b"
+#define DUAL_BOOT_FLAG_REGEX	"\\bendless\\.image\\.device\\b"
+
+#define NORMAL_BOOT	0x0
+#define DUAL_BOOT	0x1
+#define LIVE_BOOT	0x2
+
+/*
+ * emer_boot_id_provider_get_boot_type:
+ *
+ * Get boot type by checking kernel's boot command line.
+ *
+ * Returns: a number 0x0 for normal boot, 0x1 for dual boot, 0x2 for live boot.
+ */
+guchar
+emer_boot_id_provider_get_boot_type (void)
```

Comment:
The original [`EmerBootIdProviderPrivate`](https://github.com/endlessm/eos-event-recorder-daemon/blob/edcb6abcccc33aaf33d1908065d71deb318c3055/daemon/emer-persistent-cache.c#L53) is only used/related by `EmerPersistentCachePrivate` in `daemon/emer-persistent-cache.c`, not `daemon/emer-daemon.c`.  It is hard and complicated to get the **boot_type** via `daemon/emer-persistent-cache.c` in `daemon/emer-daemon.c`.

The **boot_type** is good to be cached, but is it good to be a property in `EmerBootIdProviderPrivate`?

Reply:
Not needed here, but I think what you'd do is add an instance of `EmerBootIdProvider` to `EmerDaemonPrivate` like the other providers. Then you would use the `get_boot_type` method during the request construction. In the `get_boot_type` implementation you'd use a `static` variable like in `emer_boot_id_provider_get_id` so that you only read it once store the result in `EmerBootIdProviderPrivate`.

For the persistent cache you could just have another instance of `EmerBootIdProvider` or you could pass it in when calling `emer_persistent_cache_new` from `emer_daemon_constructed`. `emer_persistent_cache_new_full` even supports passing in the boot ID provider already.

But anyways it's a bit premature optimization.

Reply:
Okay!  Thanks